runtime/server/queries_charts.go

```diff
@@ -0,0 +1,116 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/drivers"
+	"github.com/rilldata/rill/runtime/pkg/httputil"
+	"github.com/rilldata/rill/runtime/server/auth"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func (s *Server) GetChartData(w http.ResponseWriter, req *http.Request) error {
+	// TODO: telemetry
+
+	ctx := req.Context()
+	instanceId := req.PathValue("instance_id")
+	chartName := req.PathValue("chart_name")
+	// TODO: is a separate auth needed?
+	if !auth.GetClaims(ctx).CanInstance(instanceId, auth.ReadMetrics) {
+		return ErrForbidden
```

Comment:
This returns grpc error which I think will map to a bad request. Should use `httputil.Error`.